src/ts/Search/Location.tsx

```diff
@@ -27,18 +27,25 @@ export default class Location extends React.Component<any, any> {
 
   componentDidMount() {
     // in case of re-creating component we do not need another script
-    if (document.getElementById('google-maps-script')) {
-      this.initMap();
-      return;
+    if (typeof google !== 'undefined') {
```

Comment:
Perhaps it's a slight exaggeration but http://stackoverflow.com/questions/3390396/how-to-check-for-undefined-in-javascript